test/integration/targets/nxos_gir/tests/common/sanity.yaml

```diff
@@ -94,26 +86,23 @@
   - name: "Remove snapshots"
```

Comment:
Let's add one more set of steps to the cleanup section in this test.  I have seen cases where a switch gets stuck in maintenance mode and simply issuing the `no system mode maintenance` command will not work because the command is not even recognized as a valid command.  The following sequence seems to always work (at least in my tests)

1. `configure maintenance profile normal-mode`
2. `no system mode maintenance`

Reply:
This one may not work in failure scenarios, the command "system mode maintenance" takes 120+ seconds to get into maintenance mode in working cases. If you hit timeout or an error with "system mode maintenance" command and it has not been 120+ seconds, the switch would not let you take it out of maintenance mode. We may have to add bit of delay before trying to clean up in failure scenarios.

Reply:
Good points.  I am starting to lean towards adding code to make the module itself more robust.  We should update the action plugin for nxos set the timeout automatically for this module instead of forcing the user to set a timeout that is large enough.  Here is an example of how I am warning users to set the timeout to a value large enough when upgrading images using the `nxos_install_os` module.

https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/action/nxos.py#L67

In this case for the  `nxos_gir` module we could just set it to say 200 seconds to allow enough time for the maintenance mode to complete and then add logic in the module to handle any timeout that is beyond 200 seconds to try and roll back to the previous state.